shared/metrics/metrics_test.go

```
@@ -0,0 +1,66 @@
+	"strings"
+	"testing"
+	"time"
+)
+
+// waitForServer polls addr until it accepts connections or the deadline passes.
+func waitForServer(t *testing.T, url string) *http.Response {
+	t.Helper()
+	deadline := time.Now().Add(2 * time.Second)
+	var lastErr error
+	for time.Now().Before(deadline) {
+		resp, err := http.Get(url)
```

golangci-lint (2.12.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)